torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py

```diff
@@ -316,9 +322,18 @@ def collect_bw_donated_buffer_idxs(
     bw_outs = next(reversed(bw_module.graph.find_nodes(op="output"))).args[0]
     fw_outs = next(reversed(fw_module.graph.find_nodes(op="output"))).args[0]
 
-    fw_ins = [n.meta["val"] if hasattr(n, "meta") else None for n in fw_ins]
-    fw_outs = [n.meta["val"] if hasattr(n, "meta") else None for n in fw_outs]
-    bw_outs = [n.meta["val"] if hasattr(n, "meta") else None for n in bw_outs]
+    fw_ins = [
+        n.meta["val"] if (hasattr(n, "meta") and "val" in n.meta) else None
```

Comment:
Did you see actually witness any `fw_ins` or `fw_outs` that were `None` during testing?

I think `bw_outs` can potentially be None (if we've determined statically that certain inputs do not need gradients), but I wouldn't expect this for `fw_ins/outs`. If so I'd rather only have the none check for the backward outs, to make the invariants clearer.

Reply:
oh right i misread, thanks